### DISCUSSION TITLE
Hide subfolder navigation but don't exclude the folders

### DISCUSSION THREAD

**Discussion by atomicbird** [obsidian-html/obsidian-html#435] [Ideas]
It would be nice if I could hide subfolders from the navigation list in the left pane without excluding the folders via `exclude_subfolders`. The scenario is:

- I have inline images in many Obsidian notes
- The images are normally kept in a subfolder named `assets` instead of in the same folder as the note
- I'm using `process_all: True` since starting from the entry point doesn't find all of my notes.
- As a result, `assets` appears in my HTML output

I can't exclude `assets` and still have the images, but it would be great to have some way to keep it out of the HTML folder list.

It's a minor thing at worst but this would be a really good future update for me.

**Comment by dwrolvink** [comment]
I believe this is already possible, unless I understand you wrong.

Take for example: https://obsidian-html.github.io . Note that there is no folder `Resources` in the root, even though I have that folder in the root of my vault (and use notes from it).

Note that path `Resources/` is not in `exclude_subfolders`: https://github.com/obsidian-html/obsidian-html.github.io/blob/7c032dcac417db4e6ef80470bc7c3f1061e6d0d2/__src/config.yml#L60 
(This setting just blocks any file from a path from being included. And by default has paths such as `.git`, `.obsidian`, etc.)

Compare that with `toggles/features/create_index_from_dir_structure/exclude_subfolders`: 
https://github.com/obsidian-html/obsidian-html.github.io/blob/7c032dcac417db4e6ef80470bc7c3f1061e6d0d2/__src/config.yml#L247
This setting does contain `Resources`, and it also has another setting below it to exclude certain file paths as well.

So if you don't have a setting in your config yet for `toggles/features/create_index_from_dir_structure`, copy the block below, and adjust the paths for your usecase. That should solve your issue.

``` yaml
toggles:
  features:
    create_index_from_dir_structure:
      enabled: True
      exclude_subfolders:
        - ".git"
        - "__src"
        - "md"
        - "obs.html"
        - "assets"
      exclude_files:
        - "favicon.ico"
        - "not_created.html"
```

> Default settings file: https://github.com/obsidian-html/obsidian-html/blob/master/obsidianhtml/src/defaults_config.yml

Note that wildcards, glob patterns, etc, are not supported. So at the moment you have to add 1 line for each assets folder that you want to exclude. If I understand you correctly you have multiple of these folders. 

It should be relatively easy to add something like a glob pattern support to exclude **all** asset folders in one go. If you could use this, please make an issue for it, then I will add it in the next release.

**Reply by atomicbird** [reply]
Thanks, that's exactly what I needed, I just had not found it in the docs yet. I've added #436 for globbing in this field.
